### PR TITLE
added CED tests to GIN

### LINKED ISSUE #303: GIN tests gap: Mac OS X for sonpy
## Description

When we added CED tests to our `test_gin.py` collection, the CI revealed some weird issue with importing the `sonpy` package seen here: https://github.com/catalystneuro/nwb-conversion-tools/runs/4358475697?check_suite_focus=true

The runner reports the package installed correctly, without issue, but still raises the lazy check for that interface only, indicating some other `ImportError`.

For now, it's simply manually disabled in the `test_gin` to ignore if running on mac.

## Steps to Reproduce

I don't have quick immediate access to a Mac device to test this right at the moment, so anyone that does is welcome to try

```
conda create -n test_sonpy python=3.7  # since that's what our CI runs  on
conda activate test_sonpy
# from nwb-conversion-tools
pip install -r requirements-full.txt
pip install -e .

python
> import sonpy
```

## Environment

Master branch of nwb-conversion-tools, full CI runner specification (`requirements-full.txt`).
    

## Checklist

- [X] Have you ensured the feature or change was not already [reported](https://github.com/catalystneuro/nwb-conversion-tools/issues)?
- [X] Have you included a brief and descriptive title?
- [X] Have you included a clear description of the problem you are trying to solve?
- [X] Have you included a minimal code snippet that reproduces the issue you are encountering?
- [X] Have you checked our [Contributing](https://nwb-conversion-tools.readthedocs.io/en/master/developer_guide.html) document?

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
fix #303 

## Motivation

Recently, the neo team added `.smrx` files to the testing repo. This adds the tests to our repo, which requires some fixes to get them to pass.

## How to test the behavior?
```
pytest test_gin.py
```

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

File: nwb_conversion_tools/datainterfaces/ecephys/ced/ceddatainterface.py
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ecephys/ced/ceddatainterface.py:31]:
```suggestion
        """Retrieve and inspect necessary channel information prior to initialization."""
```

Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/ecephys/ced/ceddatainterface.py:31]: I really need to find a spell check for linting in my IDE; every other type of format check has been supported so far, lol.

**Comment by bendichter** [comment]
@CodyCBakerPhD I am fine with this solution for now, as long as you can open an issue about the osx issue for us to come back to later

**Comment by CodyCBakerPhD** [comment]
> @CodyCBakerPhD I am fine with this solution for now, as long as you can open an issue about the osx issue for us to come back to later

Sounds good, wanted to check if it actually works first. Then I'll raise. Apparently `neo` bypasses this (and presumably other issues) by not even running mac in their matrix.

**Comment by bendichter** [comment]
Good thing we test OSX then!

**Comment by bendichter** [comment]
related to https://github.com/catalystneuro/nwb-conversion-tools/issues/303

**Comment by CodyCBakerPhD** [comment]
@bendichter Good to go on my end - giving a special message in the case of that Mac + Python version, and added tests for that assertion as well in the CI.

**Comment by CodyCBakerPhD** [comment]
@luiztauffer If you could take a look at this since Ben is busy

File: tests/test_interfaces.py
Comment (luiztauffer) [inline tests/test_interfaces.py:42]: @CodyCBakerPhD to be sure, the tests are not running for MacOS so far, right?
Is this test case ever running? I couldn't find the logs for it in the tests and the if/else seems to raise an error or skip it with a message

File: tests/test_gin.py
Comment (luiztauffer) [inline tests/test_gin.py:111]: ok, so from here I understand we're not testing with MacOS, right? The tests Python v is 3.7
is this intended?

we can probably get rid of the `class test(TestCase)` in `tests/test_interfaces.py` then?

**Comment by luiztauffer** [comment]
@CodyCBakerPhD `test_gin.py` doesn't seem to be running?

File: tests/test_gin.py
Comment (CodyCBakerPhD) [reply tests/test_gin.py:111]: MacOS does run full tests, yes (when GIN data available). See https://github.com/catalystneuro/nwb-conversion-tools/pull/307 for v3.8 and 3.9

File: tests/test_interfaces.py
Comment (CodyCBakerPhD) [reply tests/test_interfaces.py:42]: It should be on this branch; this branch runs only 3.7 so should skip on mac but pass on windows full and linux full, even without GIN data. I'll see if I can find it in CI logs

**Comment by CodyCBakerPhD** [comment]
@luiztauffer Yes, #310 is a priority to ensure that. Can wait until that gets merged and update this to check

**Comment by CodyCBakerPhD** [comment]
@luiztauffer However, you can indeed still confirm that without #310, the GIN tests did run successfully on one of the previous iterations (https://github.com/catalystneuro/nwb-conversion-tools/runs/4373747307?check_suite_focus=true); everything after that was an update from master to stay up to date and shouldn't affect CED at all.

If you want to wait, I'm meeting with Alessio to go over #310 tomorrow so should be merged by 5pm UTC

**Review by luiztauffer — APPROVED** [review]

**Pull request merged**
